workshops/management/commands/check_certificates.py

```diff
@@ -19,15 +20,18 @@ def handle(self, *args, **options):
         path_to_root = options['path']
 
         badges = self.get_badges()
+        result = [['which','badge','username','person','email','awarded']]
         for (name, badge) in badges:
-            db_people = self.get_db_people(badge)
+            db_records = self.get_db_people(badge)
+            db_people = db_records.keys()
             cert_path = os.path.join(path_to_root, name)
             if not os.path.isdir(cert_path):
-                print('No directory {0}'.format(name))
+                print('No directory {0}'.format(name), file=sys.stderr)
             else:
                 file_people = self.get_file_people(cert_path)
-                self.report_missing('database but not disk', name, db_people - file_people)
-                self.report_missing('disk but not database', name, file_people - db_people)
+                self.missing(result, 'database-disk', name, db_people - file_people, db_records)
+                self.missing(result, 'disk-database', name, file_people - db_people, db_records)
```

Comment:
`db_people` is a set created with `db_records.keys()`

On 2016-02-24 10:11 AM, Piotr Banaszkiewicz wrote:
>
> In workshops/management/commands/check_certificates.py 
> <https://github.com/swcarpentry/amy/pull/708#discussion_r53951759>:
>
> >              else:
> >                  file_people = self.get_file_people(cert_path)
> > -                self.report_missing('database but not disk', name, db_people - file_people)
> > -                self.report_missing('disk but not database', name, file_people - db_people)
> > +                self.missing(result, 'database-disk', name, db_people - file_people, db_records)
> > +                self.missing(result, 'disk-database', name, file_people - db_people, db_records)
>
> Does this work? Can you really do |dict() - set()| or |set() - dict()|?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/swcarpentry/amy/pull/708/files#r53951759>.
>

-- 
Dr Greg Wilson
Director of Instructor Training
Software Carpentry Foundation